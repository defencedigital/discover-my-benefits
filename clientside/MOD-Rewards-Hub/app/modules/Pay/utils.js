export const isOF5AndAbove = rangeFound => rangeFound && (rangeFound.PayRangeCode === 'OF5' || rangeFound.PayRangeCode === 'OF6' || rangeFound.PayRangeCode === 'OF7' || rangeFound.PayRangeCode === 'OF8' || rangeFound.PayRangeCode === 'OF9');
