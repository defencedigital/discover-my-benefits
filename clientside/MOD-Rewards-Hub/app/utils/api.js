import axios from 'axios';

// Tried to externalise with process.env but doesn't seem to work as expected
// Using axios as it's installed and gives a nicer interface to fetch
export const API = axios.create({
  responseType: 'json',
  headers: {
    'Content-Type': 'application/json;charset=utf-8',
    Accept: 'application/json',
  },
});
