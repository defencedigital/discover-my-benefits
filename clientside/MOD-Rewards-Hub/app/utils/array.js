const removeDuplicates = (myArr, prop) => myArr.filter((obj, pos, arr) => arr.map((mapObj) => mapObj[prop]).indexOf(obj[prop]) === pos);

const chunk = (array, groupsize) => {
  if (!(array instanceof Array)) {
    throw new Error('Parameter is not a array!');
  }

  const a = [].concat(array);

  const sets = [];
  let i = 0;

  const chunks = a.length / groupsize;

  while (i < chunks) {
    sets[i] = a.splice(0, groupsize);
    i += 1;
  }

  return sets;
};

module.exports = {
  removeDuplicates,
  chunk,
};
