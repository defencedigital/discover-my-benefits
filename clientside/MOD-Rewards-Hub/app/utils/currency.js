import currencyFormatter from 'currency-formatter';

export const formatCurrency = (value) => currencyFormatter.format(value, { code: 'GBP' });
