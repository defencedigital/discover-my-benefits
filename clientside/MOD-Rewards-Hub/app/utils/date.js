export const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const formatDate = (date, lastDayOfTheMonth) => {
  const day = (lastDayOfTheMonth) ? new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate() : date.getDate();
  const monthIndex = date.getMonth();
  const year = date.getFullYear();

  return `${day} ${monthNames[monthIndex]} ${year}`;
};
