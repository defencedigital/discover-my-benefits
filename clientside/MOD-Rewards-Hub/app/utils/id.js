let dependencyID = -999999999999;

export const getId = () => {
  dependencyID += 1;
  return dependencyID;
};
