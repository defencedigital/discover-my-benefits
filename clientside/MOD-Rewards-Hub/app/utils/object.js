const memoize = require('fast-memoize');
const transform = require('lodash').transform;
const isEqual = require('lodash').isEqual;
const isObject = require('lodash').isObject;
/* eslint-disable */

const parseDotNotation = (str, val, obj)  => {
  let currentObj = obj;
  let keys = str.split('.');
  let i;
  let l = Math.max(1, keys.length - 1);
  let key;

  for (i = 0; i < l; ++i) {
    key = keys[i];
    currentObj[key] = currentObj[key] || {};
    currentObj = currentObj[key];
  }

  currentObj[keys[i]] = val;
  delete obj[str];
};

const expandFN = (obj) => {
  for (let key in obj) {
    if (key.indexOf('.') !== -1) {
      parseDotNotation(key, obj[key], obj);
    }
  }
  return obj;
};

const expand = memoize(expandFN);

const isJSONFN = (jsonString) => {
  try {
    var o = JSON.parse(jsonString);
    if (o && typeof o === "object") {
      return o;
    }
  } catch (e) { }

  return false;
};

const isJSON = memoize(isJSONFN);

/**
 * Deep diff between two object, using lodash
 * @param  {Object} object Object compared
 * @param  {Object} base   Object to compare with
 * @return {Object}        Return a new object who represent the diff
 */
function difference(object, base) {
  return transform(object, (result, value, key) => {
    if (!isEqual(value, base[key])) {
      result[key] = isObject(value) && isObject(base[key]) ? difference(value, base[key]) : value;
    }
  });
}

module.exports = {
  expand,
  isJSON,
  difference,
};
