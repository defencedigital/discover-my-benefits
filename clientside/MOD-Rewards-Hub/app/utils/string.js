/* eslint-disable */

/**
 * A collection of useful string utilities
 */

/**
 * Converts a string to url friendly string
 * @param string
 *
 * Replace spaces with -
 * Remove all non-word chars
 * Replace multiple - with single -
 * Trim - from start of text
 * Trim - from end of text
 *
 */

const slugify = (string) =>
  string.toString()
  .toLowerCase()
  .replace(/\s+/g, '-')
  .replace(/[^\w\-]+/g, '')
  .replace(/\-\-+/g, '-')
  .replace(/^-+/, '')
  .replace(/-+$/, '');

module.exports = {
  slugify,
};
