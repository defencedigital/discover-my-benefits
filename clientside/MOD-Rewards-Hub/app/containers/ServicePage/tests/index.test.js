import React from 'react';
import { fromJS } from 'immutable';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ServicePage } from '../index';
import { shallowWithIntl } from '../../../utils/intl-enzyme-test-helper';
import { service, category } from '../../../tests/commonMocks';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  onChangeLocale: () => {},
  onChangeTheme: () => {},
  id: service.id,
  intl: {},
  service,
  categories: [category],
  services: [service],
  benefits: [],
  allQuestions: [],
  questions: [],
  options: [],
  dependencies: [],
  calculations: [],
  allBenefits: [],
  allCategories: [],
  commitmenttypes: [],
  fsCalculations: [],
});

describe('ServicePage component', () => {
  const MountedServicePage = shallowWithIntl(<ServicePage {...mockedProps.toJS()} />);

  it('matches the snapshot', () => {
    const html = MountedServicePage;
    expect(html).toMatchSnapshot();
  });
});
