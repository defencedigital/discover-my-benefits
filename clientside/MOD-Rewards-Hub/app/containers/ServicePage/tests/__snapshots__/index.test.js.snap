// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicePage component matches the snapshot 1`] = `ShallowWrapper {}`;
